shared/src/main/java/dice/StatsProcessor.java

```diff
@@ -161,42 +161,44 @@ private StatsData computeStats(RollRecord[] records) {
         double avgRolls = 0.0;
         double avgScore = 0.0;
         double avgNumDiceUsed = 0.0;
-
-        for (int index = 0; index < dbLength; index++) {
-            int temp = 0;
-            RollRecord record = records[index];
-            totalGames = record.getGameId();
-            totalDiceUsed += record.getNumDice();
-        }
-
-        avgNumDiceUsed = (double)totalDiceUsed / (double)dbLength;
-        totalGames = totalGames + 1;
-        avgRolls = (double)totalRolls / (double)totalGames;
-
         int maxGameScore = 0;
-        if (dbLength > 1) {
-            for (int index = 1; index < dbLength; index++) {
-                int[] tempArray = new int[totalGames];
+
+        if (dbLength != 0) {
```

Comment:
Can you add unit tests (either to `TestStatsProcessor` or `TestMockStatsProcessor`) for this new path? Normally I like to add the test before the feature and make sure that it breaks. Then adding the feature should fix the test. You can do it retroactively though, just make sure the test breaks when you comment out this code.